### PR TITLE
refactor left items to use Menu component

### PR DESCRIPTION
**Pull request by benschac**
## What does this PR do and why?

refactor menu items to use new API.
## Screenshots or screen recordings

_If applicable, provide screenshots or screen recordings to demonstrate the changes._

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [ ] I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

File: src/components/Navbar/MenuItems.tsx
Comment (tomquirk) [inline src/components/Navbar/MenuItems.tsx:130]: **question (non-blocking)** wondering if we can just use ExternalLink here instead?

**Review by tomquirk — COMMENTED** [review]
One comment!

File: src/components/Navbar/MenuItems.tsx
Comment (benschac) [reply src/components/Navbar/MenuItems.tsx:130]: The thought was it's a little easier to parse. Its a bit more explicit, and you know we're using a `next/link`. Honestly not married to it, just thought it could simplify things.

Comment (tomquirk) [reply src/components/Navbar/MenuItems.tsx:130]: Hmm, IMO ExternalLink is a simpler (and safer) approach here, because it means the developer doesn't have to remember to add externalMenuLinkProps every time.

I don't feel too strongly though. Approving.

**Review by tomquirk — APPROVED** [review]

**Comment by benschac** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2097?utm_source=stack-comment).


Job Reference: OhSplG4sLw5ccXyOTEUT

**Comment by benschac** [comment]
**Graphite Merge Job**

Current status: 🔄 Waiting for checks to pass

This pull request is currently waiting for CI and mergeability checks to pass before being merged.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2097?utm_source=stack-comment).


Job Reference: 5CwyltIj9iIbfOFYRnkc

**Pull request merged**
